sources/web/datalab/server.ts

```diff
@@ -155,10 +155,15 @@ function handleRequest(request: http.ServerRequest,
       (path.indexOf('/edit') == 0) ||
       (path.indexOf('/sessions') == 0)) {
 
-    if (path.indexOf('/tree') == 0) {
-        loadedSettings[startup_path_setting] = path
-        settings_.updateUserSetting(userId, startup_path_setting, path, true);
+    // When the user asks for something in the tree, we eventually get called
+    // here with a corresponding URL that starts with /api/contents/.
```

Comment:
Yeah, that makes more sense. I changed the code to check that the file exists and is a directory.